app/scripts/modules/core/src/task/task.states.ts

```diff
@@ -40,6 +43,36 @@ module(TASK_STATES, [APPLICATION_STATE_PROVIDER]).config([
       children: [taskDetails],
     };
 
+    const taskLookup: INestedState = {
+      name: 'taskLookup',
+      url: '/tasks/:taskId',
+      params: {
+        taskId: { dynamic: true },
+      },
+      redirectTo: transition => {
+        const { taskId } = transition.params();
+
+        if (!taskId) {
+          return undefined;
+        }
+
+        return Promise.resolve()
```

Comment:
nit: Could this be shortened to `return Promise.resolve(TaskReader.getTask(taskId))`?

Reply:
Yup. Thanks! Dunno why I did that..